### ISSUE TITLE
Проблема со схемой с микрофоном.

### ISSUE THREAD

**Issue by rfdrew** [AlexGyver/ColorMusic#1]
Может быть я такой везучий, но при работе схемы с микрофоном нашел баг. При первом режиме работы, загорается и горит пол ленты. Когда крутишь резистор добавляешь или уменьшаешь количество светодиодов которые горят. Потом постепенно количество светящихся светодиодов приходит в норму. Калибровка шума не помогает. Лечится добавлением резистора, я ставил 10-30 кОм(Больше не было) все приходит в норму. Немного страдает чувствительность, но на 30 КОм она уже не значительная.
Проверьте и добавьте себе в проект это изменение. Если нужно могу видео предоставить с объяснениями.
PS: Используется Arduino UNO, все остальное как в проекте.

**Comment by CharlzVard** [comment]
Решается заменой типа byte на int в строках 116 и 236.

**Comment by AlexGyver** [comment]
В версии 2.6.1 это исправлено

**Comment by kurtveliev** [comment]
> Может быть я такой везучий, но при работе схемы с микрофоном нашел баг. При первом режиме работы, загорается и горит пол ленты. Когда крутишь резистор добавляешь или уменьшаешь количество светодиодов которые горят. Потом постепенно количество светящихся светодиодов приходит в норму. Калибровка шума не помогает. Лечится добавлением резистора, я ставил 10-30 кОм(Больше не было) все приходит в норму. Немного страдает чувствительность, но на 30 КОм она уже не значительная.
> Проверьте и добавьте себе в проект это изменение. Если нужно могу видео предоставить с объяснениями.
> PS: Используется Arduino UNO, все остальное как в проекте.

**Добрый** день! У меня такая же проблема на новой прошивке 2.10. Схема с микрофоном, не работают первые 2 режима. Пожалуйста подскажите какие именно строки нужно поменять?**

**Comment by kurtveliev** [comment]
> Решается заменой типа byte на int в строках 116 и 236. (версия 2.6, по крайней мере)

Какие именно строки? в прошивке 2.10 я так понимаю этой проблемы быть не должно, но у меня все же первые два режима не работают должным образом

**Comment by CharlzVard** [comment]
> Какие именно строки? в прошивке 2.10 я так понимаю этой проблемы быть не должно, но у меня все же первые два режима не работают должным образом

А открыть  версию 2.6 и посмотреть?

116: uint16_t LOW_PASS = 100;
236: byte low_pass;

**Comment by kurtveliev** [comment]
Cпасибо, но не помогло(

**Comment by CharlzVard** [comment]
Калибровку делали?

**Comment by kurtveliev** [comment]
да, что только не делал. И на другие аналоговые пины перекидывал, и потенциометром пробовал, и вручную в прошивке пробовал подобрать. И конденсатор менял. Единственное не пробовал другой резистор на LED. Но дело в том что с третего режима все прекрасно работает, а первые два просто плавно горит часть диодов и никакого движения

**Comment by DaniilGlubochenko** [comment]
Друзья, тоже столкнулся с это проблемой. Смена типа в 116 и 236 строках не помогла. Также работают все режимы, кроме 1-го и 2-го. Пробовал и на прошивке 2.10 и на 2.6. 

Подскажите, пожалуйса, как можно решить?

**Comment by pitybull** [comment]
и у меня тоже эта проблема . Первые два режима не работают . Собрал Схему с микрофонам  Версия 2.10
что делать? 
@AlexGyver  помоги пожалуйста

**Comment by DaniilGlubochenko** [comment]
> и у меня тоже эта проблема . Первые два режима не работают . Собрал Схему с микрофонам Версия 2.10
> что делать?
> @AlexGyver помоги пожалуйста

Привет!

Добрался я до вот этого обсуждения.
https://community.alexgyver.ru/threads/cvetomuzyka-na-arduino-obsuzhdenie-proekta.126/

Там в постах есть несколько решений этой проблемы. Полистай, не помню на какой именно страничке. Я пока что еще ничего не пробовал. 

Отпишись плиз о результатах потом.

**Comment by pitybull** [comment]
Я решил проблемы с первым и вторим режимом.
Прочитал все 47 страниц. 
https://community.alexgyver.ru/threads/cvetomuzyka-na-arduino-obsuzhdenie-proekta.126/
avalonix на 25 странице и viktor001 пару раз писали что делать. 
avalonix впаял 45k Резистор viktor001 10к Резистор

viktor001 объяснил это так  
https://community.alexgyver.ru/threads/cvetomuzyka-na-arduino-obsuzhdenie-proekta.126/page-44#post-21834

> Если у вас помимо синего фона, без сигнала на входе, светятся какое то количество пикселей, то это говорит о наличие потенциала на этом самом входе. Грубо говоря напряжение, которого не должно быть. Для этого в схему внесён дополнительный резистор между А2 и минусом.

Я впаял Резистор 10к между А2 и GND на arduino, откалибровал звук и всё заработало. 

![Screenshot_1](https://user-images.githubusercontent.com/60965033/74675216-5bd67a80-51b3-11ea-9c89-b854b539a633.jpg)

**Comment by kurtveliev** [comment]
> Я решил проблемы с первым и вторим режимом.
> Прочитал все 47 страниц.
> https://community.alexgyver.ru/threads/cvetomuzyka-na-arduino-obsuzhdenie-proekta.126/
> avalonix на 25 странице и viktor001 пару раз писали что делать.
> avalonix впаял 45k Резистор viktor001 10к Резистор
> 
> viktor001 объяснил это так
> https://community.alexgyver.ru/threads/cvetomuzyka-na-arduino-obsuzhdenie-proekta.126/page-44#post-21834
> 
> > Если у вас помимо синего фона, без сигнала на входе, светятся какое то количество пикселей, то это говорит о наличие потенциала на этом самом входе. Грубо говоря напряжение, которого не должно быть. Для этого в схему внесён дополнительный резистор между А2 и минусом.
> 
> Я впаял Резистор 10к между А2 и GND на arduino, откалибровал звук и всё заработало.
> 
> ![Screenshot_1](https://user-images.githubusercontent.com/60965033/74675216-5bd67a80-51b3-11ea-9c89-b854b539a633.jpg)

Спасибо большое, на досуге буду пробовать. Я насчет резистора смекнул, но впаял его последовательно перед кондером и ничего не помогло ( делетант в этом деле)

**Comment by kurtveliev** [comment]
впаял на 20ком по совету выше и все заработало! Спасибо!

**Comment by DaniilGlubochenko** [comment]
> Я решил проблемы с первым и вторим режимом.
> Прочитал все 47 страниц.
> https://community.alexgyver.ru/threads/cvetomuzyka-na-arduino-obsuzhdenie-proekta.126/
> avalonix на 25 странице и viktor001 пару раз писали что делать.
> avalonix впаял 45k Резистор viktor001 10к Резистор
> 
> viktor001 объяснил это так
> https://community.alexgyver.ru/threads/cvetomuzyka-na-arduino-obsuzhdenie-proekta.126/page-44#post-21834
> 
> > Если у вас помимо синего фона, без сигнала на входе, светятся какое то количество пикселей, то это говорит о наличие потенциала на этом самом входе. Грубо говоря напряжение, которого не должно быть. Для этого в схему внесён дополнительный резистор между А2 и минусом.
> 
> Я впаял Резистор 10к между А2 и GND на arduino, откалибровал звук и всё заработало.
> 
> ![Screenshot_1](https://user-images.githubusercontent.com/60965033/74675216-5bd67a80-51b3-11ea-9c89-b854b539a633.jpg)

Гран мерси за инфу! На выходных доберусь до радиорынка и тоже буду пробовать.
